js/app.js

```diff
@@ -248,6 +209,12 @@ function getNoResultsMessage() {
 }
 
 
+function getUrlParamater(param) {
+    var urlParams = new URLSearchParams(window.location.search);
+    return urlParams.has(param) ? urlParams.get(param) : undefined;
+}
+
```

Comment:
There are a few ways to refine this.
1. Unless you particularly need to return `undefined` as opposed to null, you can just do `return urlParams.get(param)`
2. The fact that you will be constructing `URLSearchParams` each time this function is called is suboptimal unless you are planning on changing `window.location.search` between calls to this function. Perhaps it would be better to parse `location.href` into a `URL` object at the very top of this script, so that you can use it whenever you need to throughout.

Reply:
Do you mean like this:

`const pageUrl = new URL(location.href);`

Throws an error when attempting to access it:

Uncaught ReferenceError: Cannot access 'pageUrl' before initialization

Reply:
So here we have an interesting quirk of JavaScript called hoisting: https://scotch.io/tutorials/understanding-hoisting-in-javascript

Consider whether or not the pageUrl variable is being initialised before you call this function

Reply:
Assuming you fixed this